docs/adr/0003-decouple-welcome-email-from-new-user-flow.md

```diff
@@ -0,0 +1,78 @@
+<!-- ADR template, source: https://github.com/adr/madr -->
+
+# Decouple welcome email from new user flow
+
+- Status: Proposed
+- Deciders: Luke Crouch, Tony Cinotto
+- Date: 2023-09-05
+
+Technical Story: [MPP-3328](https://mozilla-hub.atlassian.net/browse/MPP-3328)
+
+## Context and Problem Statement
+
+To reduce the response time from when the user interacts with the UI on Firefox for [MPP-3257](https://mozilla-hub.atlassian.net/browse/MPP-3257),
+Relay added a welcome email for new users that gets processed via `send_first_email` receiver that looks for the `user_signed_up` signal. For Phase 2 of Relay integration to Firefox, we added a `/api/v1/terms-accepted-user/` endpoint to create new users with the [resource flow](OAUTH_RESOURCE_SERVER) instead of relying party flow. A typical response from POST request to `/api/v1/terms-accepted-user/`` ranges from 1000-1200ms. The response with the welcome email processing removed is 700-900ms. The maximum time saved from separating the welcome email could be as high as 300ms. Meanwhile we need to ensure that a new user who joins Relay gets welcome email.
+
+To improve response time for `/api/v1/terms-accepted-user/` and provide reliable welcome email to new users joining Relay, how should the welcome email be processed?
+
+[OAUTH_RESOURCE_SERVER]: https://www.oauth.com/oauth2-servers/the-resource-server/
+
+## Decision Drivers
+
+- Reduce response time for `/api/v1/terms-accepted-user/`
+- Send welcome emails to new Relay users
+- Quickest implementation
+
+## Considered Options
+
+1. Implement [Redis Queue](REDIS_QUEUE) to send welcome email
+2. Utilize [Basket API](BASKET) to send welcome email
+3. Implement cron job to send welcome email
+
+## Decision Outcome
+
+Proceeded with **Option 3. Implement cron job to send welcome email** because it had the quickest implementation and possibility for reuse for Option 1. Implemented on PR [#3750](https://github.com/mozilla/fx-private-relay/pull/3750).
+
+### Positive Consequences
+
+- Welcome email is sent more reliably
+- Faster new user created on both the normal Relay flow and the Firefox Integration flow
```

Comment:
suggestion (non-blocking): another tip from hemingwayapp: use active voice rather than passive voice. e.g., say "Relay sends welcome emails more reliably" rather than "Welcome email is sent more reliably" and say "Relay creates all new user objects faster" rather than "Faster new user created on both the normal Relay flow and the Firefox Integration flow"

Reply:
I was following the previous styles on the other ADRs in our repo. I don't swing one way or the other but whatever we decide on let's update the ADR template.

Reply:
https://github.com/mozilla/fx-private-relay/pull/3870

Reply:
However, I do think it is helpful to use active voices when discussing the pros and cons on the options which I applied in [a710d2](https://github.com/mozilla/fx-private-relay/pull/3865/commits/a710d287a3d46b30ff5e34fa048fc61781592884)